apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java

```diff
@@ -123,61 +132,87 @@ public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
-                    '0');
-            // ndcProductColumns[3] - Proprietary Name
-            // ndcProductColumns[13] - Substance Name
-            ndcProductHashMap.put(
-                String.format("%s-%s", nationalDrugCodeManufacturer, nationalDrugCodeIngredient),
-                ndcProductColumns[3] + " - " + ndcProductColumns[13]);
-          } catch (StringIndexOutOfBoundsException e) {
-            continue;
-          }
-        }
-      } catch (IOException e) {
-        throw new UncheckedIOException("Unable to read NDC code data.", e);
-      }
-    }
+  protected Map<String, String> readFDADrugCodeFile(
+      boolean includeFakeDrugCode, InputStream inputStream) {
+
+    ndcProductHashMap = getFdaProcessedData(includeFakeDrugCode, inputStream);
+
     if (includeFakeDrugCode) {
       ndcProductHashMap.put(FAKE_DRUG_CODE, FAKE_DRUG_CODE_DISPLAY);
     }
     return ndcProductHashMap;
   }
+
+  /**
+   * Reads all the <code>PRODUCTNDC</code> and <code>SUBSTANCENAME</code> fields from the FDA NDC
+   * Products file which was downloaded during the build process.
+   *
+   * <p>See {@link gov.cms.bfd.data.fda.utility.App} for details.
+   *
+   * @param includeFakeDrugCode is whether to incliude fake drug code
+   * @param inputStream is the inputStream that is passed in
+   * @return a map with drug codes and fields.
+   */
+  protected Map<String, String> getFdaProcessedData(
+      boolean includeFakeDrugCode, InputStream inputStream) {
+
+    Map<String, String> ndcProcessedData = new HashMap<String, String>();
+
+    try (final InputStream ndcProductStream = inputStream;
+        final BufferedReader ndcProductsIn =
+            new BufferedReader(new InputStreamReader(ndcProductStream))) {
+      /*
+       * We want to extract the PRODUCTNDC and PROPRIETARYNAME/SUBSTANCENAME from the FDA Products
+       * file (fda_products_utf8.tsv is in /target/classes directory) and put in a Map for easy
+       * retrieval to get the display value which is a combination of PROPRIETARYNAME &
+       * SUBSTANCENAME
+       */
+      String line = "";
+      ndcProductsIn.readLine();
+      while ((line = ndcProductsIn.readLine()) != null) {
+        String ndcProductColumns[] = line.split("\t");
+        String nationalDrugCodeManufacturer =
+            StringUtils.leftPad(
+                ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].substring(
+                    0, ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].indexOf("-")),
+                5,
+                '0');
+        String nationalDrugCodeIngredient =
+            StringUtils.leftPad(
+                ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].substring(
+                    ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].indexOf("-") + 1,
+                    ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].length()),
+                4,
+                '0');
+        // ndcProductColumns[3] - Proprietary Name
+        // ndcProductColumns[13] - Substance Name
```

Comment:
Probably dont need this with our new index names!

Reply:
took out comments of fied names